frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/PipelineBuilderForm.tsx

```diff
@@ -52,10 +52,19 @@ const PipelineBuilderForm: React.FC<PipelineBuilderFormProps> = (props) => {
     setFieldValue,
     setStatus,
     values,
+    validateForm,
   } = props;
   const statusRef = React.useRef(status);
   statusRef.current = status;
 
+  React.useEffect(() => {
+    if (values.editorType === EditorType.Form) {
+      // Force validation against the new data that was adjusted in the YAML
+      // Formik isn't properly handling the immediate state of the form values during the cycle of the editorType
+      setTimeout(() => validateForm(), 0);
+    }
+  }, [values.editorType, validateForm]);
```

Comment:
Why not use `useFormikValidationFix` hook here? It does a similar hack.

Reply:
I tried using this hook, but unfortunately it doesn't work.  with pipeline builder form component, validation happens before the task is preselected, so the above code is to validate the form right after all the new changes are in place, thus setTimeout is used.